catalog/views.py

```diff
@@ -42,6 +42,36 @@ def show_course(request, slug: str):
         .order_by("-edited")
     )
 
+    if request.user.is_authenticated:
```

Comment:
Could we try to have a bit less code duplication here ? 
I'm guessing something like this could work: (untested)

```python
if not request.user.is_authenticated:
    display = Q(hidden=False)
else:
    if request.user.is_staff:
        display = Q()
    else:
        display = Q(hidden=True, user=request.user) | Q(hidden=False)


documents = (
  course.document_set.exclude(state="ERROR")
  .filter(display)
  .select_related("course", "user")
  .prefetch_related("tags", "vote_set")
  .annotate(upvotes=Count("vote", filter=Q(vote__vote_type=Vote.VoteType.UPVOTE)))
  .annotate(
      downvotes=Count("vote", filter=Q(vote__vote_type=Vote.VoteType.DOWNVOTE))
  )
  .order_by("-edited")
)
```

Reply:
Way smarter ! I updated the code